src/Files.App/Services/Storage/StorageDevicesService.cs

```diff
@@ -22,8 +22,20 @@ public async IAsyncEnumerable<ILocatableFolder> GetDrivesAsync()
 			var googleDrivePath = App.AppModel.GoogleDrivePath;
 			var pCloudDrivePath = App.AppModel.PCloudDrivePath;
 
+#if DEBUG
+			Debug.WriteLine($"In RDS.GDA: googleDrivePath: {googleDrivePath}");
+#endif
+
 			foreach (var drive in list)
 			{
+				var shouldSkip = false;
+				await foreach (var cloudProvider in GoogleDriveCloudDetector.GetGoogleDriveProvidersFromRegistryAsync(false))
```

Comment:
I’m a bit worried about the potential performance impact this change might have. That’s actually one of the reasons we initially didn’t display Google Drive in the cloud section.

Reply:
It looks like that code takes about a quarter of a second to run on my machine in debug mode, and it only runs once. I just pushed a commit with some timing in place so you can experiment with it if you want.

Output of `Get-ComputerInfo -Property "CsProscessors"` on my machine, for reference:

```
CsProcessors
------------
{Intel(R) Core(TM) i7-8700T CPU @ 2.40GHz}
```

Screenshot of the timing:

![timing](https://github.com/user-attachments/assets/3f7c5cd2-2fd8-4ee7-b7a2-6aa0f62bc01d)

Reply:
Doesn't it depend on the number of drives?

Reply:
Trial 1:

In RDS.GDA after registry check: Time elapsed for filter: 00:00:01.7072234
In RDS.GDA: drive.Name: C:\

In RDS.GDA after registry check: Time elapsed for filter: 00:00:00.6069444
In RDS.GDA: drive.Name: G:\

Trial 2:

In RDS.GDA after registry check: Time elapsed for filter: 00:00:01.6830505
In RDS.GDA: drive.Name: C:\

In RDS.GDA after registry check: Time elapsed for filter: 00:00:00.4277025
In RDS.GDA: drive.Name: G:\

Trial 3:

In RDS.GDA after registry check: Time elapsed for filter: 00:00:02.0122619
In RDS.GDA: drive.Name: C:\

In RDS.GDA after registry check: Time elapsed for filter: 00:00:00.8921746
In RDS.GDA: drive.Name: G:\

Trial 4:

In RDS.GDA after registry check: Time elapsed for filter: 00:00:01.9693083
In RDS.GDA: drive.Name: C:\

In RDS.GDA after registry check: Time elapsed for filter: 00:00:00.3338732
In RDS.GDA: drive.Name: G:\

Not great! I'll look into how I can improve this.

Reply:
> Doesn't it depend on the number of drives?

Yes it seems like it does. I was mistaken at first.

Reply:
Another option is to remove the drives after the app is fully loaded, but this is not ideal.

Reply:
I just pushed a change that improves the performance of that registry check quite a bit, please let me know if it looks alright. Sorry there are so many changes. I basically just split up some of the methods I wrote so I could reuse the smallest essential part of the code that gets the registry path.

Here are the trials I ran, all in Debug mode:

Trial 1

In RemovableDrivesService: Time elapsed for registry check: 00:00:00.0770886

Trial 2

In RemovableDrivesService: Time elapsed for registry check: 00:00:00.0609413

Trial 3

In RemovableDrivesService: Time elapsed for registry check: 00:00:00.0727057

Trial 4

In RemovableDrivesService: Time elapsed for registry check: 00:00:00.0918816

Reply:
Also it only checks the registry once now. I set `App.AppModel.GoogleDrivePath` in `RemovableDrivesService` now, since that's where we need the path first. And then GoogleDriveCloudDetector reads from there, since that gets called second.

Reply:
This looks a lot better

Reply:
Excellent, thanks.

Reply:
Should I click "Resolve conversation" on here?